pkg/chart/ChartService.go

```diff
@@ -1720,3 +1750,135 @@ func (impl ChartServiceImpl) GetCustomChartInBytes(chartRefId int) ([]byte, erro
 	}
 	return manifestByteArr, nil
 }
+
+func (impl ChartServiceImpl) UpdateGitRepoUrlInCharts(appId int, chartGitAttribute *util.ChartGitAttribute, userId int32) error {
+	charts, err := impl.chartRepository.FindActiveChartsByAppId(appId)
+	if err != nil && pg.ErrNoRows != err {
+		return err
+	}
+	for _, ch := range charts {
+		if ChartsUtil.IsGitOpsRepoNotConfigured(ch.GitRepoUrl) {
```

Comment:
what if allow custom repository is false in global gitops, we need to check it here also?

Reply:
It has been verified prior to the function call